### PR TITLE
Minor warning fixes

### PR DESCRIPTION
**Pull request by spevans**
Reduce the noise from excessive warnings. Most of these are based on the fixits emitted by the compiler

### CODE REVIEW

File: Foundation/CharacterSet.swift [hunk 1]
```diff
@@ -486,7 +486,7 @@ public struct CharacterSet : ReferenceConvertible, Equatable, Hashable, SetAlgeb
     
     @_semantics("convertToObjectiveC")
     public func _bridgeToObjectiveC() -> NSCharacterSet {
-        return unsafeBitCast(_wrapped, to: NSCharacterSet.self)
+        return _wrapped
```
Comment (parkera) [inline Foundation/CharacterSet.swift]: @phausler did swift-corelibs-foundation get the new CharacterSet implementation yet?

File: Foundation/Measurement.swift [hunk 2]
```diff
@@ -75,7 +75,7 @@ extension Measurement where UnitType : Dimension {
 /// Add two measurements of the same Unit.
 /// - precondition: The `unit` of `lhs` and `rhs` must be `isEqual`.
 /// - returns: A measurement of value `lhs.value + rhs.value` and unit `lhs.unit`.
-public func +<UnitType : Unit>(lhs: Measurement<UnitType>, rhs: Measurement<UnitType>) -> Measurement<UnitType> {
+public func +<UnitType>(lhs: Measurement<UnitType>, rhs: Measurement<UnitType>) -> Measurement<UnitType> {
```
Comment (parkera) [inline Foundation/Measurement.swift]: Did we make these changes in the Swift overlay too?

File: Foundation/NSKeyedArchiver.swift [hunk 3]
```diff
@@ -612,7 +612,7 @@ open class NSKeyedArchiver : NSCoder {
 
             if _isContainer(object) {
                 guard let codable = object as? NSCoding else {
-                    fatalError("Object \(object) does not conform to NSCoding")
+                    fatalError("Object \(String(describing:object)) does not conform to NSCoding")
```
Comment (parkera) [inline Foundation/NSKeyedArchiver.swift]: Nit pick: space after the `:` please.

File: Foundation/ReferenceConvertible.swift [hunk 4]
```diff
@@ -11,7 +11,7 @@
 /// Decorates types which are backed by a Foundation reference type.
 ///
 /// All `ReferenceConvertible` types are hashable, equatable, and provide description functions.
-public protocol ReferenceConvertible : CustomStringConvertible, CustomDebugStringConvertible, Hashable, Equatable {
```
Comment (parkera) [inline Foundation/ReferenceConvertible.swift]: Seems obnoxious that this generates a warning, but ok.

File: Foundation/NSKeyedArchiver.swift (hunk 3 above)
Comment (spevans) [inline Foundation/NSKeyedArchiver.swift]: fixed

File: Foundation/Measurement.swift (hunk 2 above)
Comment (spevans) [inline Foundation/Measurement.swift]: Looks like it was done as part of https://github.com/apple/swift/pull/8281

File: Foundation/CharacterSet.swift (hunk 1 above)
Comment (spevans) [inline Foundation/CharacterSet.swift]: @parkera Is this the only outstanding issue with this PR? I can always revert this one change if so

Comment (parkera) [inline Foundation/CharacterSet.swift]: Let's leave this one out for now.

Comment (spevans) [inline Foundation/CharacterSet.swift]: Ive reverted this change

**Comment by parkera** [comment]
@swift-ci test and merge

**Comment by pushkarnk** [comment]
I think the build failure was later fixed by a PR yesterday.

@swift-ci test and merge

**Pull request merged**